packages/core/src/codecs/text-codec.ts

```diff
@@ -91,7 +91,14 @@ export default class TextCodec implements ContentCodec {
         if (be) {
             return Buffer.from(body, be);
         } else {
-            return Buffer.from(body);
+            if (this.subMediaType === "image/svg+xml" && typeof value === "string") {
```

Comment:
Question: Maybe we can add an option in the TextCodec constructor whether wrapping is needed or not. That makes it more generic. What do you think?

Reply:
Created `bufferDirectWrite` options since it is needed for HTML, SVG etc as well..